tools/lint-sh.js

```diff
@@ -0,0 +1,139 @@
+#!/usr/bin/env node
+'use strict';
+
+const { spawn } = require('child_process');
+const { promises: fs, readdirSync } = require('fs');
+const { extname, join, relative, resolve } = require('path');
+const DIR = resolve(process.argv[2]);
+
+const SHELLCHECK_EXE = process.env.SHELLCHECK || '/usr/local/bin/shellcheck';
+const GIT_EXE = process.env.GIT || '/usr/bin/git';
```

Comment:
Should probably assume these exist on `PATH` and output some helpful message if they are not.

Reply:
> For shellcheck, there's also a npm module to download it: https://github.com/gunar/shellcheck

Yeah, when I need to run it, I use `npx`: `npx shellcheck [args]`